components/datadog/agent/kubernetes_helm.go

```diff
@@ -129,7 +129,7 @@ func NewHelmInstallation(e config.CommonEnvironment, args HelmInstallationArgs,
 		linuxInstallName += "-linux"
 	}
 
-	values := buildLinuxHelmValues(installName, agentImagePath, agentImageTag, clusterAgentImagePath, clusterAgentImageTag, randomClusterAgentToken.Result, !args.DisableLogsContainerCollectAll)
+	values := buildLinuxHelmValues(installName, agentImagePath, agentImageTag, clusterAgentImagePath, clusterAgentImageTag, randomClusterAgentToken.Result, args.DisableLogsContainerCollectAll)
```

Comment:
I thought so too, but after experimenting with the initial configuration, I found that the previous method set `container_collect_all` to true in the agent when `WithoutLogsContainerCollectAll` was applied, whereas with this configuration `container_collect_all` is correctly set to off.

Reply:
I think it was because I missed this: https://github.com/DataDog/test-infra-definitions/pull/814
The `disableContainerCollectAll` was never passed so no matter if you used `WithoutLogsContainerCollectAll` the default value was passed to `buildLinuxHelmValues` (ie `!false`) and I think that's why it was always enabled

Reply:
Ohh okay gotcha, thanks for the clarification.